docs/fundamentals/code-analysis/style-rules/ide0306.md

```diff
@@ -0,0 +1,88 @@
+- IDE0306
+dev_langs:
+- CSharp
+---
+# Use collection expression for new (IDE0306)
+
+| Property                 | Value                                         |
+|--------------------------|-----------------------------------------------|
+| **Rule ID**              | IDE0306                                       |
+| **Title**                | Use collection expression for new             |
+| **Category**             | Style                                         |
+| **Subcategory**          | Language rules (expression-level preferences) |
+| **Applicable languages** | C# 12+                                        |
+| **Options**              | `dotnet_style_prefer_collection_expression`   |
+
+## Overview
+
+This rule flags places where a [collection expression](../../../csharp/language-reference/operators/collection-expressions.md) with a spread element (`..`) could be used to initialize a list instead of `new`. For example, this rule offers to simplify code like `new List<...>(x)` into the collection expression form (`[.. x]`).
+
+## Options
+
+Options specify the behavior that you want the rule to enforce. For information about configuring options, see [Option format](language-rules.md#option-format).
+
+### dotnet_style_prefer_collection_expression
+
+| Property                 | Value                                     | Description                           |
+|--------------------------|-------------------------------------------|---------------------------------------|
+| **Option name**          | dotnet_style_prefer_collection_expression |                                       |
+| **Option values**        | `true` &#124; `when_types_exactly_match`  | Prefer to use collection expressions only when types match exactly, for example, `List<int> m1 = new List<int>(new[] { 1, 2, 3 });`. |
+|                          | `when_types_loosely_match`<sup>\*</sup>   | Prefer to use collection expressions even when types match loosely, for example, `IEnumerable<int> m1 = new List<int>(new[] { 1, 2, 3 });`. The targeted type must match the type on the right-hand side or be one of the following types: <xref:System.Collections.Generic.IEnumerable%601>, <xref:System.Collections.Generic.ICollection%601>, <xref:System.Collections.Generic.IList%601>, <xref:System.Collections.Generic.IReadOnlyCollection%601>, <xref:System.Collections.Generic.IReadOnlyList%601>. |
+|                          | `false` &#124; `never`                    | Disables the rule.                    |
+| **Default option value** | `when_types_loosely_match`                |                                       |
+
+<sup>\*</sup>When this option is used, the code fix might change the semantics of your code.
+
+## Example
+
+```csharp
+// Code with violation.
+List<int> l1 = new List<int>(Enumerable.Range(1, 10));
+List<int> m1 = new List<int>(new[] { 1, 2, 3 });
+
+// Fixed code.
+List<int> l1 = [.. Enumerable.Range(1, 10)];
+List<int> m1 = [.. new[] { 1, 2, 3 }];
```

Comment:
OK, so the fixed code still looks bad to me. I'd prefer:

```csharp
List<int> m1 = [ 1, 2, 3 ];
```

I know that is semantically different, so it might be worth stating that the rule won't make that additional change.

Reply:
we could consider making the fixer make that change.  or having another fixer for that.